### PR TITLE
MGDAPI-409 Added alert for multi-az pod distribution and link to SOP

### PR DESCRIPTION
**Pull request by CathalOConnorRH**
# Description
Added alert for multi-AZ pod distribution and SOP link

To verify, 
1. Install RHOAM on a multi-az cluster. 
2. when cluster is installed, update local aws cli credential details with AWS details for the cluster. 
3. Run `./scripts/disableAZ.sh true us-east-1c` to disable the `us-east-1c` zone. Change details if needed depending on where the cluster is installed. 
4. Wait for all pods to be recreated in the remaining zones. This can take some time and the console may go offline for a period. 
5. Once pods are recreated, re-enable the zone you disabled `./scripts/disableAZ.sh false us-east-1c` 
6. Check the status of the nodes under `compute` and when all are ready, check for firing alert. 
7. Follow SOP to confirm redistribution of pods to the newly enabled zone.

### CODE REVIEW

File: pkg/products/monitoring/prometheusRules.go [hunk 1]
```diff
@@ -299,6 +299,23 @@ func (r *Reconciler) newAlertsReconciler() resources.AlertReconciler {
...
+				Namespace: r.Config.GetOperatorNamespace(),
+				GroupName: "general.rules",
+				Rules: []monitoringv1.Rule{
+					{
+						Alert: "PodDistribution",
+						Annotations: map[string]string{
+							"sop_url": resources.SopUrlPodDistributionIncorrect,
+							"message": "Pod {{  $labels.namespace  }} / {{  $labels.pod  }} ({{  $labels.container  }}) is incorretly distributed to the zone {{  $value  }} ; for the last 5 minutes",
+						},
+						Expr:   intstr.FromString("kube_pod_info{namespace=~\"redhat-rhoam-.*\", created_by_kind!=\"<none>\"} <= on (namespace, created_by_name) group_left (label_topology_kubernetes_io_zone) (count by (namespace, created_by_name, label_topology_kubernetes_io_zone) (kube_pod_info{namespace=~\"redhat-rhoam-.*\", created_by_kind!=\"<none>\"} == on (node) group_left  (label_topology_kubernetes_io_zone) kube_node_labels) > on (namespace, created_by_name) group_left ceil(count by (namespace, created_by_name) (kube_pod_info{namespace=~\"redhat-rhoam-.*\", created_by_kind!=\"<none>\"} == on (node) group_left  (label_topology_kubernetes_io_zone) kube_node_labels) / scalar(count(count by (label_topology_kubernetes_io_zone) (kube_node_labels)))))"),
```
Comment (obrienrobert) [inline pkg/products/monitoring/prometheusRules.go:313]: Could we remove the hard-coded ns prefix (`redhat-rhoam`) and use the `nsPrefix` var instead?

Comment (CathalOConnorRH) [reply pkg/products/monitoring/prometheusRules.go]: Updated now. I would love you input on if this is the best place to put this alert or if you think there would be somewhere better?

File: pkg/products/monitoring/prometheusRules.go [hunk 2]
```diff
@@ -299,6 +299,23 @@ func (r *Reconciler) newAlertsReconciler() resources.AlertReconciler {
 					},
 				},
 			},
+			{
+				AlertName: "pod-distribution",
```
Comment (obrienrobert) [inline pkg/products/monitoring/prometheusRules.go:303]: Could the alert name be more descriptive? Maybe something like `MultiAZPodDistribution` or `RhoamMultiAZPodDistribution` etc.

Comment (CathalOConnorRH) [reply pkg/products/monitoring/prometheusRules.go]: Updated now. Thanks

**Comment by obrienrobert** [comment]
@CathalOConnorRH I haven't had a chance to run through the verification steps, but I'm wondering how this alert performs on a non a multi-AZ cluster?

**Pull request merged**